classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -488,4 +495,51 @@ protected function setFallbackVersion(): ?string
         }
         return null;
     }
+
+    /**
+     * Check the contexts' contact details before upgrade
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function checkContactSetting(): void
+    {
+        $primaryLocale = DB::table('site')
+            ->select(['primary_locale'])
+            ->first()
+            ->primary_locale;
+
+        $missingContactContexts = DB::table("{$this->getContextTable()}")
+            ->select([
+                "{$this->getContextKeyField()}", 
+                "enabled", 
+            ])
+            ->addSelect(["title" => DB::table("{$this->getContextSettingsTable()}")
+                ->select(["setting_value"])
+                ->whereColumn("{$this->getContextKeyField()}", "{$this->getContextTable()}.{$this->getContextKeyField()}")
+                ->where("locale", $primaryLocale)
+                ->where("setting_name", "name")
+                ->take(1)
```

Comment:
Probably, we don't need to limit the result here, I think it will always return 1 row with a context name as the combination of journal_id, locale and setting_name is unique

Reply:
right, removed .